aten/src/ATen/native/cuda/ScatterGatherKernel.cu

```diff
@@ -288,7 +289,41 @@ void scatter_fill_cuda_kernel(Tensor& self, int64_t dim, const Tensor& index, Sc
   );
 }
 
+template <bool is_complex_t>
+struct scatter_add_op {
+  template <typename scalar_t>
+  C10_DEVICE void operator()(scalar_t* lhs, const scalar_t* rhs);
+};
+
+template<>
+struct scatter_add_op</*is_complex_t=*/true> {
```

Comment:
let's add test for complex dtypes to verify scatter_add works correctly for complex

Reply:
Done. I also wanted to add complex test for gather.backward (uses scatter_add), but realized that the gather port was not yet merged...